File: src/neoforge/service/main/java/settingdust/mod_sets/neoforge/util/LoaderAdapter.java
package settingdust.mod_sets.neoforge.util;

import net.neoforged.fml.loading.FMLLoader;
import net.neoforged.fml.loading.FMLPaths;
import settingdust.mod_sets.ModSets;

import java.nio.file.Path;

public class LoaderAdapter implements settingdust.mod_sets.util.LoaderAdapter {

    public static final Path configDirectory = FMLPaths.CONFIGDIR.get().resolve(ModSets.ID);

    public static boolean inGame = false;

    @Override
    public boolean isClient() {
        return FMLLoader.getDist().isClient();
    }

    @Override
    public boolean isModLoaded(final String modId) {
        return FMLLoader.getLoadingModList().getModFileById(modId) != null;
    }

    @Override
    public Path getConfigDirectory() {
        return configDirectory;
    }

    @Override
    public Path getModsDirectory() {
        return FMLPaths.MODSDIR.get();
    }

    @Override
    public boolean isInGame() {
        return inGame;
    }
}

File: src/neoforge/1.21/main/kotlin/settingdust/mod_sets/neoforge/util/LoaderAdapter.kt
package settingdust.mod_sets.neoforge.util

import net.neoforged.fml.loading.FMLLoader
import net.neoforged.fml.loading.FMLPaths
import net.neoforged.fml.loading.LoadingModList
import settingdust.mod_sets.ModSets
import settingdust.mod_sets.util.LoaderAdapter
import java.nio.file.Path
import kotlin.io.path.div

class LoaderAdapter : LoaderAdapter {
    override val isClient: Boolean
        get() = FMLLoader.getDist().isClient

    override fun isModLoaded(modId: String) = LoadingModList.get().getModFileById(modId) != null

    override val configPath by lazy { FMLPaths.CONFIGDIR.get() / ModSets.ID }

    override val modsDir: Path = FMLPaths.MODSDIR.get()
}
